### PR TITLE
Feat/bearer mw

### PR DESCRIPTION
**Pull request by stefanprokopdev**

### CODE REVIEW

File: src/lib/authMiddleware.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+};
+
+export const expressBearer = (options: AuthistOptions) => async (
+    request: Request & { user?: User },
+    response: Response,
+    next: NextFunction
+) => {
+    const [error, user] = await bearer(options, request.headers.authorization);
+    if (error) {
```
Comment (smoliji) [inline src/lib/authMiddleware.ts:33]: I have a strong feeling that this branch implementation is too much specific to be hardcoded here.

The json and the error serialization especially, e.g. not showing any error information on production.

I suggest to keep this as a default behaviour for a new authist option `onExpressAuthenticationError` (or whatever :D) to allow the user
- serialize error and close the response
- use the `next` function to process the error in express way

File: src/test/integration/bearer.test.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                userModel.create({ email: data.email, password: hashedPassword }),
+        },
+    });
+    const email = 'tester@test.com';
+    const password = '***';
+    beforeAll(async () => {
+        userModel = await initDb();
+        const authenticator = getAuthenticator(getAuthistOptions());
+        const server = createServer(authenticator);
+        server.listen(3000);
```
Comment (smoliji) [inline src/test/integration/bearer.test.ts]: (optional) There is a better way to do this than using specific port - use port: 0. This way the server starts on any available port the OS has to offer. There should be a way to access it from express too, it's `server.address().port` on the net.Server instance if you are using native.

https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback


The huge benefit besides the hardcoded port (and failing the tests when your OS is using that port) is that you can easily start multiple servers during the tests in parallel, without worrying about port clash.

**Review by smoliji — CHANGES_REQUESTED** [review]

File: src/lib/authMiddleware.ts (hunk 1 above)
Comment (stefanprokopdev) [reply src/lib/authMiddleware.ts:33]: I added `onExpressAuthenticationError` function.

File: src/test/integration/bearer.test.ts (hunk 2 above)
Comment (stefanprokopdev) [reply src/test/integration/bearer.test.ts]: I used port 0

File: src/lib/authMiddleware.ts (hunk 1 above)
Comment (stefanprokopdev) [reply src/lib/authMiddleware.ts:33]: here is the test https://github.com/AckeeCZ/authist/pull/9/files#diff-f6d6d4d8c737d88d6836bbbf0fa102fa99fe29faadd2d4af61d8031543d973eaR99

File: src/test/integration/bearer.test.ts (hunk 2 above)
Comment (smoliji) [reply src/test/integration/bearer.test.ts]: Looks good

**Review by smoliji — APPROVED** [review]

**Pull request merged**